client/client_test.go

```diff
@@ -98,26 +115,104 @@ func initializeRepo(t *testing.T, rootType, tempBaseDir, gun, url string) (
 	rootPubKey, err := repo.CryptoService.Create("root", rootType)
 	assert.NoError(t, err, "error generating root key: %s", err)
 
-	err = repo.Initialize(rootPubKey.ID())
-	assert.NoError(t, err, "error creating repository: %s", err)
-
 	return repo, rootPubKey.ID()
 }
 
-// TestInitRepo runs through the process of initializing a repository and makes
-// sure the repository looks correct on disk.
-// We test this with both an RSA and ECDSA root key
-func TestInitRepo(t *testing.T) {
-	testInitRepo(t, data.ECDSAKey)
+// Initializing a new repo while specifying that the server should manage the root
+// role will fail.
+func TestInitRepositoryManagedRolesIncludingRoot(t *testing.T) {
+	// Temporary directory where test files will be created
+	tempBaseDir, err := ioutil.TempDir("/tmp", "notary-test-")
+	assert.NoError(t, err, "failed to create a temporary directory")
+	defer os.RemoveAll(tempBaseDir)
+
+	repo, rootPubKeyID := createRepoAndKey(
+		t, data.ECDSAKey, tempBaseDir, "docker.com/notary", "http://localhost")
+	err = repo.Initialize(rootPubKeyID, data.CanonicalRootRole)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(),
```

Comment:
Seems like we should create an error type for this case of attempting to have the server manage an unsupported key.